src/goOutline.ts

```diff
@@ -151,44 +139,49 @@ function convertToCodeSymbols(
 		}
 
 
-		let range = null;
-		if (document && byteOffsetToDocumentOffset) {
-			let start = byteOffsetToDocumentOffset(decl.start - 1);
-			let end = byteOffsetToDocumentOffset(decl.end - 1);
-			range = new vscode.Range(document.positionAt(start), document.positionAt(end));
-			if (decl.type === 'type') {
-				let line = document.lineAt(document.positionAt(start));
-				let regex = new RegExp(`^\\s*type\\s+${decl.label}\\s+struct\\b`);
-				decl.type = regex.test(line.text) ? 'struct' : 'type';
-			}
+		const start = byteOffsetToDocumentOffset(decl.start - 1);
+		const end = byteOffsetToDocumentOffset(decl.end - 1);
+		const startPosition = document.positionAt(start);
+		const endPosition = document.positionAt(end);
+		const symbolRange = new vscode.Range(startPosition, endPosition);
+		const selectionRange = startPosition.line === endPosition.line ?
+			symbolRange :
+			new vscode.Range(startPosition, document.lineAt(startPosition.line).range.end);
+
+		if (decl.type === 'type') {
+			let line = document.lineAt(document.positionAt(start));
+			let regex = new RegExp(`^\\s*type\\s+${decl.label}\\s+struct\\b`);
+			decl.type = regex.test(line.text) ? 'struct' : 'type';
 		}
 
-		let symbolInfo = new vscode.SymbolInformation(
+		const symbolInfo = new vscode.DocumentSymbol(
 			label,
+			decl.type,
```

Comment:
Is this the "details" that is shown next to the symbol in the outline feature?

Reply:
yes ( so in this case would look like the name and next to it the type ( e.g. package, function etc)

Reply:
also looking in go-outline, everything seems fine, but we stop only at the function level. do we own https://github.com/lukehoban/go-outline ? seems we could get more from the AST

Reply:
We own the fork of it https://github.com/ramya-rao-a/go-outline
Feel free to look into that for improvements

Reply:
well it seems we might be able to get more info from the AST, of things inside the functions, @jhendrixMSFT has more experience with the Go AST, he can confirm.

Reply:
Absolutely.  I've opened https://github.com/Microsoft/vscode-go/issues/2370 to track enhancing the outline view.